### PR TITLE
Document around docker.sock

### PR DESCRIPTION
**Pull request by plaindocs**
Going to merge this, and look at the second half another day.

### CODE REVIEW

File: pages/agent/v3/gcloud.md.erb [hunk 1]
```diff
@@ -76,25 +76,37 @@ spec:
     metadata:
       labels:
         app: buildkite-agent
+      annotations:
+        "cluster-autoscaler.kubernetes.io/safe-to-evict": "true"
```
Comment (yob) [inline pages/agent/v3/gcloud.md.erb:80]: We can probably leave this annotation out. Some folks might want it, but it's not essential and it has some sharp edges. Maybe it belongs follow up to the getting-started shaped guide?

File: pages/agent/v3/gcloud.md.erb [hunk 2]
```diff
@@ -177,19 +189,20 @@ spec:
...
       mountPath: /root/.ssh/id_rsa
       subPath: id_rsa
-    - name: ssh-keys
+    - name: buildkite-agent-ssh
       mountPath: /root/.ssh/id_rsa.pub
       subPath: id_rsa.pub
     ...
   volumes:
-  - name: ssh-keys
+  - name: buildkite-agent-ssh
```
Comment (yob) [inline pages/agent/v3/gcloud.md.erb:200]: the name of these volumes can be anything, so it'd be fine to keep them as `ssh-keys` if you were keen to keep the diff of this PR minimal.

Comment (plaindocs) [reply pages/agent/v3/gcloud.md.erb]: Thanks @yob !

**Comment by plaindocs** [comment]
@yob I've made minor changes to text and updated screenshots, and I'm happy to merge if you are.

File: pages/agent/v3/gcloud.md.erb [hunk 3]
```diff
@@ -78,58 +79,71 @@ spec:
...
-
-  _           _ _     _ _    _ _                                _
- | |         (_) |   | | |  (_) |                              | |
- | |__  _   _ _| | __| | | ___| |_ ___    __ _  __ _  ___ _ __ | |_
- | '_ \| | | | | |/ _` | |/ / | __/ _ \  / _` |/ _` |/ _ \ '_ \| __|
- | |_) | |_| | | | (_| |   <| | ||  __/ | (_| | (_| |  __/ | | | |_
- |_.__/ \__,_|_|_|\__,_|_|\_\_|\__\___|  \__,_|\__, |\___|_| |_|\__|
-                                                __/ |
- http://buildkite.com/agent                    |___/
-2017-02-16 03:34:10 NOTICE Starting buildkite-agent v3.0.0 with PID: 5
```
Comment (yob) [inline pages/agent/v3/gcloud.md.erb:124]: 3.0.0! :scream:

**Pull request merged**